python/cudf/cudf/core/column/string.py

```diff
@@ -5938,15 +5930,15 @@ def view(self, dtype) -> "cudf.core.column.ColumnBase":
         str_end_byte_offset = self.base_children[0].element_indexing(
             self.offset + self.size
         )
-        char_dtype_size = self.base_children[1].dtype.itemsize
+        char_dtype_size = cudf.api.types.dtype("int8").itemsize
```

Comment:
**issue (throughout)**: Please don't hardcode `"int8"` as the dtype for string column characters everywhere. There must be a data buffer we can obtain it from?

Reply:
data buffer is of type `cudf.core.buffer.Buffer`, it does not hold dtype.

Reply:
Ah, ok. Can we introduce (like `cudf._lib.types.size_type_dtype`) a single source of truth for the type of the string char buffer, perhaps `cudf._lib.types.char_type_dtype`?

Reply:
`int8` is used only in this place. Ideally we would want to get rid of it, since `int8` is actually a workaround to hold byte data buffer in strings column. In follow up PRs, we will eliminate references of `int8` related to character buffer.

Reply:
Here the`int8` indicates a byte data buffer as column. It is not related to `char` type.

Reply:
I think it would be worth adding a `# TODO` comment noting that int8 is a workaround

Reply:
removed int8 here.